backend/src/providers/transactions.ts

```diff
@@ -263,3 +299,86 @@ export const collectNestedReceiptWithOutcome = (
     },
   };
 };
+
+export const getDeposit = (actions: Action[]) =>
+  actions
+    .map((action: Action) => {
+      if ("deposit" in action.args) {
+        return BigInt(action.args.deposit);
+      } else {
+        return 0n;
+      }
+    })
+    .reduce((accumulator, deposit) => accumulator + deposit, 0n)
+    .toString();
```

Comment:
I believe we should yield `BigInt`s here and convert to string the last mile - before sending values to the client.